ingestion/src/metadata/data_insight/processor/cost_analysis_report_data_processor.py

```diff
@@ -0,0 +1,234 @@
+
+"""
+Processor class used to compute refined report data
+"""
+
+from __future__ import annotations
+
+import traceback
+from collections import defaultdict
+from typing import Iterable
+
+from metadata.data_insight.processor.data_processor import DataProcessor
+from metadata.generated.schema.analytics.reportData import ReportData, ReportDataType
+from metadata.generated.schema.analytics.reportDataType.aggregatedCostAnalysisReportData import (
+    AggregatedCostAnalysisReportData,
+)
+from metadata.generated.schema.analytics.reportDataType.rawCostAnalysisReportData import (
+    RawCostAnalysisReportData,
+)
+from metadata.generated.schema.entity.data import table
+from metadata.utils.logger import data_insight_logger
+from metadata.utils.time_utils import get_end_of_day_timestamp_mill
+
+logger = data_insight_logger()
+
+
+class RawCostAnalysisReportDataProcessor(DataProcessor):
+    """Processor class used as a bridge to refine the data"""
+
+    _data_processor_type = "RawCostAnalysisReportData"
+
+    def fetch_data(self) -> Iterable[table.Table]:
+        try:
+            yield from self.metadata.list_all_entities(
+                table.Table, limit=1000, fields=["*"]
+            )
+        except Exception as err:
+            logger.error(f"Error trying to fetch entity -- {err}")
+            logger.debug(traceback.format_exc())
+
+    def refine(self) -> Iterable[ReportData]:
+        """Aggregate data
+        Returns:
+            list:
+        """
+
+        for entity in self.fetch_data():
+            try:
+                cost_analysis_data = RawCostAnalysisReportData(
+                    entity=self.metadata.get_entity_reference(
+                        entity=type(entity), fqn=entity.fullyQualifiedName
+                    )
+                )
+                if entity.lifeCycle:
+                    cost_analysis_data.lifeCycle = entity.lifeCycle
+
+                table_profile = self.metadata.get_latest_table_profile(
+                    fqn=entity.fullyQualifiedName
+                )
+                if table_profile.profile:
+                    cost_analysis_data.sizeInByte = table_profile.profile.sizeInByte
+
+                if cost_analysis_data.lifeCycle or cost_analysis_data.sizeInByte:
+                    yield ReportData(
+                        timestamp=self.timestamp,
+                        reportDataType=ReportDataType.RawCostAnalysisReportData.value,
+                        data=cost_analysis_data,
+                    )
+
+                self.processor_status.scanned(entity.name.__root__)
+            except Exception as err:
+                logger.debug(traceback.format_exc())
+                logger.error(f"Error trying fetch cost analysis data -- {err}")
+
+    def process(self) -> Iterable[ReportData]:
+        yield from self.refine()
+
+    def get_status(self):
+        return self.processor_status
+
+
+class AggregatedCostAnalysisReportDataProcessor(DataProcessor):
```

Comment:
both classes could come from the same parent, since at least the `fetch_data` is always going to be the same